pkg/blocks/fetchuri.go

```diff
@@ -31,144 +31,35 @@ import (
 	"github.com/facebookincubator/ttpforge/pkg/logging"
 	"github.com/spf13/afero"
 	"go.uber.org/zap"
-	"gopkg.in/yaml.v3"
 )
 
 // FetchURIStep represents a step in a process that consists of a main action,
 // a cleanup action, and additional metadata.
 type FetchURIStep struct {
-	*Act        `yaml:",inline"`
-	FetchURI    string     `yaml:"fetch_uri,omitempty"`
-	Retries     string     `yaml:"retries,omitempty"`
-	Location    string     `yaml:"location,omitempty"`
-	Proxy       string     `yaml:"proxy,omitempty"`
-	Overwrite   bool       `yaml:"overwrite,omitempty"`
-	CleanupStep CleanupAct `yaml:"cleanup,omitempty,flow"`
-	FileSystem  afero.Fs   `yaml:"-,omitempty"`
+	actionDefaults `yaml:"-"`
+	FetchURI       string   `yaml:"fetch_uri,omitempty"`
+	Retries        string   `yaml:"retries,omitempty"`
+	Location       string   `yaml:"location,omitempty"`
+	Proxy          string   `yaml:"proxy,omitempty"`
+	Overwrite      bool     `yaml:"overwrite,omitempty"`
+	FileSystem     afero.Fs `yaml:"-,omitempty"`
 }
 
 // NewFetchURIStep creates a new FetchURIStep instance and returns a pointer to it.
 func NewFetchURIStep() *FetchURIStep {
-	return &FetchURIStep{
-		Act: &Act{
-			Type: StepFetchURI,
-		},
-	}
-}
-
-// UnmarshalYAML decodes a YAML node into a FetchURIStep instance. It uses
-// the provided struct as a template for the YAML data, and initializes the
-// FetchURIStep instance with the decoded values.
-//
-// **Parameters:**
-//
-// node: A pointer to a yaml.Node representing the YAML data to decode.
-//
-// **Returns:**
-//
-// error: An error if there is a problem decoding the YAML data.
-func (f *FetchURIStep) UnmarshalYAML(node *yaml.Node) error {
```

Comment:
All of the steps have this same changeset unless otherwise noted - remove unnecessary UnmarshalYAML, remove methods no longer needed due to streamline interface